weave-js/src/components/Panel2/ChildPanelExportReport/ReportSelection.tsx

```diff
@@ -0,0 +1,156 @@
+import * as w from '@wandb/weave/core';
+import React, {useEffect} from 'react';
+
+import {useNodeValue} from '../../../react';
+import {Select} from '../../Form/Select';
+import {Icon} from '../../Icon';
+import {ChildPanelFullConfig} from '../ChildPanel';
+import {
+  customEntitySelectComps,
+  customReportSelectComps,
+} from './customSelectComponents';
+import {EntityOption, ReportOption, useEntityAndProject} from './utils';
+import {Button} from '../../Button';
+
+type ReportSelectionProps = {
+  rootConfig: ChildPanelFullConfig;
+  selectedEntity: EntityOption | null;
+  selectedReport: ReportOption | null;
+  setSelectedEntity: (entity: EntityOption) => void;
+  setSelectedReport: (report: ReportOption | null) => void;
+};
+
+export const ReportSelection = ({
+  rootConfig,
+  selectedEntity,
+  selectedReport,
+  setSelectedEntity,
+  setSelectedReport,
+}: ReportSelectionProps) => {
+  const {entityName} = useEntityAndProject(rootConfig);
+
+  // Get all of user's entities
+  const entitiesMetaNode = w.opMap({
+    arr: w.opUserEntities({user: w.opRootViewer({})}),
+    mapFn: w.constFunction({row: 'entity'}, ({row}) => {
+      return w.opDict({
+        name: w.opEntityName({entity: row}),
+        isTeam: w.opEntityIsTeam({entity: row}),
+      } as any);
+    }),
+  });
+  const entities = useNodeValue(entitiesMetaNode);
+
+  useEffect(() => {
+    // Default initial entity value based on url
+    if (!entities.loading && entities.result.length > 0) {
+      const foundEntity = entities.result.find(
+        (item: EntityOption) => item.name === entityName
+      );
+      setSelectedEntity(foundEntity);
+    }
+  }, [entityName, entities, setSelectedEntity]);
```

Comment:
Tried to fix default entity selection here...i feel like there's a more efficient way to do this? or maybe an `op` to help with this. I might iterate on this in the next PR when I introduce the project selector


https://github.com/wandb/weave/assets/103448723/7db59669-f076-49e0-b074-e6dd0e7579ce